evennia/contrib/grid/ingame_map_display/ingame_map_display.py

```diff
@@ -0,0 +1,322 @@
+
+Specifically, in `mygame/commands/default_cmdsets.py`:
+
+```
+...
+from evennia.contrib.grid.ingame_map_display import ingame_map_display  # <---
+
+class CharacterCmdset(default_cmds.Character_CmdSet):
+    ...
+    def at_cmdset_creation(self):
+        ...
+        self.add(ingame_map_display.MapDisplayCmdSet)  # <---
+
+```
+
+Then `reload` to make the new commands available.
+
+Additional Settings:
+
+In order to change your default map size, you can add to `mygame/server/settings.py`:
+
+BASIC_MAP_SIZE = 5
+
+This changes the default map width/height. 2-5 for most clients is sensible.
+
+If you don't want the player to be able to specify the size of the map, ignore any
+arguments passed into the Map command.
+"""
+import time
+from django.conf import settings
+from evennia import CmdSet
+from evennia.commands.default.muxcommand import MuxCommand
+
+_BASIC_MAP_SIZE = settings.BASIC_MAP_SIZE if hasattr(settings, 'BASIC_MAP_SIZE') else 2
+
+# _COMPASS_DIRECTIONS specifies which way to move the pointer on the x/y axes and what characters to use to depict the exits on the map.
+_COMPASS_DIRECTIONS = {
+    'north': (0, -3, ' | '),
+    'south': (0, 3, ' | '),
+    'east': (3, 0, '-'),
+    'west': (-3, 0, '-'),
+    'northeast': (3, -3, '/'),
+    'northwest': (-3, -3, '\\'),
+    'southeast': (3, 3, '\\'),
+    'southwest': (-3, 3, '/'),
+    'up': (0, 0, '^'),
+    'down': (0, 0, 'v')
+}
+
+
+class Map(object):
+    def __init__(self, caller, size=_BASIC_MAP_SIZE, location=None):
+        """
+        Initializes the map.
+
+        Args:
+            caller (object): Any object, though generally a puppeted character
+            size (int): The seed size of the map, which will be multiplied to get the final grid size
+            location (object): The location at the map's center (will default to caller.location if none provided)
+        """
+        self.start_time = time.time()
+        self.caller = caller
+        self.max_width = int(size * 2 + 1) * 5   # This must be an odd number
+        self.max_length = int(size * 2 + 1) * 3  # This must be an odd number
+        self.has_mapped = {}
+        self.curX = None
+        self.curY = None
+        self.size = size
+        self.location = location or caller.location
+
+    def create_grid(self):
+        """
+        Create the empty grid for the map based on the configured size
+
+        Returns:
+            The created grid, a list of lists
+        """
+        board = []
+        for row in range(self.max_length):
+            board.append([])
+            for column in range(int(self.max_width/5)):
+                board[row].extend([' ', '   ', ' '])
+        return board
+
+    def exit_name_as_ordinal(self, ex):
+        """
+        Get the exit name as a compass direction if possible
+
+        Args:
+            ex (Exit): The current exit being mapped
+        Returns:
+            exit_name (String): The exit name as a compass direction or an empty string
```

Comment:
The format for a `Return:` statement is `type: Text.`. In the past, Evennia was a bit lax with that, but with the move to full API-doc generation, this would lead to an error.